app/models/event.rb

```diff
@@ -85,6 +85,23 @@ def calculate_round_robin
     save
   end
 
+  def calculate_double_round_robin
```

Comment:
If I see correctly, this method is never called and so doesn't actually do anything. It should probably be checked what mode is active in the generate_schedule method and create the right pairings.